Allow user to change location of OSF folder

barbour-em opened the pull request:
Before, the Change button on the general settings page didn't do anything. Now it allows updates to the field in the text box and saves.

chennan47 commented:
![screen shot 2016-02-01 at 2 58 48 pm](https://cloud.githubusercontent.com/assets/4974056/12729367/5e21699a-c8f4-11e5-9e96-927eb81cf361.png)
After change the folder and click on change button and then hit the close of the setting panel. This dialog pops out which only supposed to pop up when you have unsaved project to sync changes. And also change button may need a rename or some other action to inform user it is taking effect.

After you change the folder location and choose to sync file again, it will think the project is empty and then remove all the files under the project on OSF.

barbour-em commented:
@chennan47, my solution for this was to sync the checked nodes after the folder location is changed. I also added a prompt to make sure the user knows what could happen. It is also more obvious what the button does.

chennan47 commented:
@barbour-em This change folder button's naming is not ideal. Usually user will expect click that button to browser through the folder to select one to change instead of a 'confirm change folder' button as what it really does. Besides the UI is broken.
![screen shot 2016-02-03 at 11 25 29 am](https://cloud.githubusercontent.com/assets/4974056/12788793/0f2b0302-ca69-11e5-90bf-2ef18aff0ddd.png)


And still a more serious issue. If I choose to sync project A and afterwards, I choose to change folder and then sync project A again. As the folder path is changed however all the files renames in old folders. It thinks you have removed all the files in your local, as a result when you click 'sync now' it will remove all the files under project A from OSF. So actually when you change the folder path, you also need to copy all the selected syncs nodes files into the new folder.

chennan47 commented:
Get it work once, then on Mac I got error message.
![screen shot 2016-02-08 at 1 44 07 pm](https://cloud.githubusercontent.com/assets/4974056/12895370/1d3ec78a-ce6a-11e5-99cd-bf4c320cad82.png)
And then nothing happen but return me back to the settings window.

On windows, Notification pops up saying error updating files which I just did a folder change which should not update the file.

The new UI interface looks neat and nice work, but since the text input route change for the folder won't do anything. It will be better to remove the input text field and let the change button take control. Also we need better notification to let user know they successfully update the folder or failed.

barbour-em commented:
I changed the text input to read only, this way the user can still see the full path to the folder. I also added notifications for success or fail of the folder copy.

As for the error, I haven't been able to make my OS X or Windows machine do this. What did you do to cause it?

chennan47 commented on osfoffline/gui/qt/preferences.py:
```diff
@@ -84,12 +89,22 @@ def set_containing_folder(self):
+        if save_setting:
+            # copy the synced folders from the old to new location
+            # so OSF doesn't think they were deleted
+            copy_tree(user.folder, os.path.join(osf_path))
```
If I put some projects which is not in the sync list or just some random folders inside the original path '/User/Desktop/OSF' for example. It will break as it can't recognize these unrelated folder which it should not copy and also if I choose the new destination to be '/User/Desktop/OSF/OSF' it will copy a new OSF folder into the current directory.

chennan47 commented on osfoffline/gui/qt/preferences.py:
```diff
@@ -84,12 +89,22 @@ def set_containing_folder(self):
+
         user.folder = os.path.join(osf_path)
 
         self.containingFolderTextEdit.setText(self._translate("Preferences", self.containing_folder))
```
Since the textedit field doesn't actually have a text edit usage. Change https://github.com/CenterForOpenScience/osf-sync/blob/develop/osfoffline/gui/qt/generated/preferences.py#L38 to user Qlabel instead of QTextedit and change the naming of the related field to avoid confusion.

abought replied:
Small aside: anything in the generated folder should probably be changed in QT Designer; otherwise the python-code changes might get lost next time someone made a change using QT Designer.

chennan47 commented:
close this in favor of https://github.com/CenterForOpenScience/osf-sync/pull/159
